src/dtx/dtx_common.c

```diff
@@ -223,19 +209,20 @@ dtx_batched_commit(void *arg)
 			}
 		}
 
+		ds_cont_child_put(cont);
+
 check:
 		if (dss_xstream_exiting(dmi->dmi_xstream))
 			break;
+
 		ABT_thread_yield();
 	}
 
 	d_list_for_each_entry_safe(dbca, tmp, &dmi->dmi_dtx_batched_list,
-				   dbca_link) {
-		if (dbca->dbca_cont->sc_cos_shutdown)
-			dtx_free_dbca(dbca);
```

Comment:
Nobody will release the ref held by 'dbca_cont'?

Reply:
It will be release via dtx_batched_commit_deregister() => dtx_free_dbca() => ds_cont_child_put().

Reply:
Yes, I mean on server shutdown we check ds_pool_child refcount to see if there was any reference leaks. There will be reference leaks on server shutdown with this patch (without cont close), but I think it can be fixed by forcing local close on shutdown in later cleanup patch.

Reply:
If the close is not closed when server shutdown, then there must be others (even without my patch) holding the reference on the container, otherwise, there will be invalid memory accessing for the ones still open the container.